File: cms-admin/src/lib/cmsSections.js
/** Keys must match backend `apps.content.cms_sections.ALL_CMS_SECTIONS`. */
export const CMS_SECTION_OPTIONS = [
  { key: "site", label: "Site & footer" },
  { key: "home", label: "Home page" },
  { key: "about", label: "About" },
  { key: "contact_page", label: "Contact page" },
  { key: "impact", label: "Impact" },
  { key: "technology", label: "Technology" },
  { key: "footer_links", label: "Footer links" },
  { key: "solutions", label: "Solutions" },
  { key: "training", label: "Training" },
  { key: "contact_submissions", label: "Form submissions" },
  { key: "upload", label: "Media upload (images)" },
];

export const NAV_ITEMS = [
  { to: "/app", end: true, label: "Overview", section: null },
  { to: "/app/site", label: "Site & footer", section: "site" },
  { to: "/app/home", label: "Home page", section: "home" },
  { to: "/app/about", label: "About", section: "about" },
  { to: "/app/contact", label: "Contact page", section: "contact_page" },
  { to: "/app/impact", label: "Impact", section: "impact" },
  { to: "/app/technology", label: "Technology", section: "technology" },
  { to: "/app/solutions", label: "Solutions", section: "solutions" },
  { to: "/app/training", label: "Training", section: "training" },
  { to: "/app/footer-links", label: "Footer links", section: "footer_links" },
  { to: "/app/submissions", label: "Form submissions", section: "contact_submissions" },
  {
    to: "/app/users",
    label: "Users",
    section: "users",
    superuserOnly: true,
  },
];

const PATH_RULES = [
  ["/app/users", "users"],
  ["/app/site", "site"],
  ["/app/home", "home"],
  ["/app/about", "about"],
  ["/app/contact", "contact_page"],
  ["/app/impact", "impact"],
  ["/app/technology", "technology"],
  ["/app/solutions", "solutions"],
  ["/app/training", "training"],
  ["/app/footer-links", "footer_links"],
  ["/app/submissions", "contact_submissions"],
];

export function sectionForPath(pathname) {
  const p = pathname.replace(/\/$/, "") || "/app";
  if (p === "/app") return null;
  for (const [prefix, sec] of PATH_RULES) {
    if (p === prefix || p.startsWith(`${prefix}/`)) return sec;
  }
  return null;
}

File: backend/apps/content/cms_sections.py
"""CMS editor section keys for fine-grained manage API access."""

from __future__ import annotations

from typing import TYPE_CHECKING, FrozenSet

if TYPE_CHECKING:
    from django.contrib.auth.models import AbstractUser

ALL_CMS_SECTIONS: FrozenSet[str] = frozenset(
    {
        "site",
        "home",
        "about",
        "contact_page",
        "impact",
        "technology",
        "footer_links",
        "solutions",
        "training",
        "contact_submissions",
        "upload",
    }
)

SECTION_LABELS: dict[str, str] = {
    "site": "Site & footer",
    "home": "Home page",
    "about": "About",
    "contact_page": "Contact page",
    "impact": "Impact",
    "technology": "Technology",
    "footer_links": "Footer links",
    "solutions": "Solutions",
    "training": "Training",
    "contact_submissions": "Form submissions",
    "upload": "Media upload (images)",
}


def get_effective_cms_sections(user: AbstractUser) -> FrozenSet[str]:
    from .models import CmsEditorProfile

    if not user or not user.is_authenticated or not getattr(user, "is_staff", False):
        return frozenset()
    if user.is_superuser:
        return ALL_CMS_SECTIONS
    try:
        prof = user.cms_profile
    except CmsEditorProfile.DoesNotExist:
        return ALL_CMS_SECTIONS
    raw = prof.sections if isinstance(prof.sections, list) else []
    return frozenset(s for s in raw if s in ALL_CMS_SECTIONS)
